modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -875,16 +936,19 @@
 			if(prob(40))
 				affected_human.try_lewd_autoemote("moan")
 			qdel(src)
-	else
-		user.visible_message(span_warning("[user] starts masturbating onto [target]!"), span_danger("You start masturbating onto [target]!"))
-		if(do_after(user, 60))
-			var/turf/target_turf = get_turf(target)
-			user.visible_message(span_warning("[user] cums on [target]!"), span_danger("You cum on [target]!"))
-			playsound(target, SFX_DESECRATION, 50, TRUE, ignore_walls = FALSE)
-			new/obj/effect/decal/cleanable/cum(target_turf)
-			if(prob(40))
-				affected_human.try_lewd_autoemote("moan")
+		return TRUE
+
+	user.visible_message(span_warning("[user] starts masturbating onto [target]!"), span_danger("You start masturbating onto [target]!"))
+	if(do_after(user, 60))
+		var/turf/target_turf = get_turf(target)
+		user.visible_message(span_warning("[user] cums on [target]!"), span_danger("You cum on [target]!"))
+		playsound(target, SFX_DESECRATION, 50, TRUE, ignore_walls = FALSE)
+		new/obj/effect/decal/cleanable/cum(target_turf)
+		if(prob(40))
+			affected_human.try_lewd_autoemote("moan")
 
-			if(target.icon_state=="stickyweb1"|target.icon_state=="stickyweb2")
-				target.icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_decals/lewd_decals.dmi'
-			qdel(src)
+		if(target.icon_state=="stickyweb1"|target.icon_state=="stickyweb2")
```

Comment:
does | even work for this

Reply:
additionally, we should not be checking icon states to determine behavior, this doesn't follow SS13 programming guidelines